pkg/engine/engine_test.go

```diff
@@ -96,3 +103,156 @@ func TestUnmarshalTaskBuild(t *testing.T) {
 		t.Errorf("Unmarshal Build task returned incorrect data")
 	}
 }
+
+func TestNewEngine(t *testing.T) {
+	env := &config.EnvConfig{
+		Daemon: config.DaemonConfig{
+			Scheduler: config.SchedulerConfig{
+				TaskRepoType:   "memory",
+				Workers:        1,
+				TaskTimeoutMin: 1,
+			},
+		},
+		Runners: map[string]config.ConfigMap{
+			"local:exec": {},
+		},
+		Builders: map[string]config.ConfigMap{
+			"exec:go": {},
+		},
+	}
+	err := env.Load()
+	require.NoError(t, err)
+
+	cfg := &EngineConfig{
+		Builders:  AllBuilders,
+		Runners:   AllRunners,
+		EnvConfig: env,
+	}
+
+	engine, err := NewEngine(cfg)
+	require.NoError(t, err)
+
+	t.Run("OK case", func(t *testing.T) {
+		testTask := createTestTask(t, "ok")
+		require.NoError(t, engine.queue.Push(testTask))
+		time.Sleep(30 * time.Second) // need to give it a time to finish because it is asynchronous function.
+		// get the task back from the database and check final state
+		taskFromDB, err := engine.store.Get(testTask.ID)
+		require.NoError(t, err)
+		require.Equal(t, task.StateComplete, taskFromDB.State().State)
+	})
+
+	t.Run("Stall case", func(t *testing.T) {
+		testTask := createTestTask(t, "stall")
+		require.NoError(t, engine.queue.Push(testTask))
+		time.Sleep(70 * time.Second) // need to give it a time to finish because it is asynchronous function.
+		// get the task back from the database and check final state
+		taskFromDB, err := engine.store.Get(testTask.ID)
+		require.NoError(t, err)
+		require.Equal(t, task.StateCanceled, taskFromDB.State().State)
+	})
+}
+
+// testcase could be one of these: 'ok' and 'stall'
+// OK will execute right away and stall will cause deadline exceed.
```

Comment:
Could you please point me to where that logic is implemented? I cannot seem to find it on my own.

Reply:
https://github.com/testground/testground/blob/master/plans/placebo/main.go
It is implemented here, I was just passing right test case as one of this from the map:
```
var testcases = map[string]interface{}{
	"ok":    ok,
	"panic": panickingTest,
	"stall": stall,
}
```
Ok returns no error
and Stall is sleeping for 24hrs